shell/platform/android/io/flutter/embedding/engine/systemchannels/AccessibilityChannel.java

```diff
@@ -0,0 +1,103 @@
+package io.flutter.embedding.engine.systemchannels;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.HashMap;
+
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.BasicMessageChannel;
+import io.flutter.plugin.common.StandardMessageCodec;
+
+/**
+ * System channel that sends accessibility requests and events from Flutter to Android.
+ * <p>
+ * See {@link AccessibilityMessageHandler}, which lists all accessibility requests and
+ * events that might be sent from Flutter to the Android platform.
+ */
+public class AccessibilityChannel {
+  @NonNull
+  public BasicMessageChannel<Object> channel;
+  @Nullable
+  private AccessibilityMessageHandler handler;
+
+  private final BasicMessageChannel.MessageHandler<Object> parsingMessageHandler = new BasicMessageChannel.MessageHandler<Object>() {
+    @Override
+    public void onMessage(Object message, BasicMessageChannel.Reply<Object> reply) {
+      // If there is no handler to respond to this message then we don't need to
+      // parse it. Return.
+      if (handler == null) {
+        return;
+      }
+
+      @SuppressWarnings("unchecked")
+      final HashMap<String, Object> annotatedEvent = (HashMap<String, Object>) message;
+      final String type = (String) annotatedEvent.get("type");
+      @SuppressWarnings("unchecked")
+      final HashMap<String, Object> data = (HashMap<String, Object>) annotatedEvent.get("data");
+
+      switch (type) {
+        case "announce":
+          String announceMessage = (String) data.get("message");
+          if (announceMessage != null) {
+            handler.announce(announceMessage);
+          }
+          break;
+        case "tap": {
+          Integer nodeId = (Integer) annotatedEvent.get("nodeId");
+          if (nodeId != null) {
+            handler.onTap(nodeId);
+          }
+          break;
+        }
+        case "longPress": {
+          Integer nodeId = (Integer) annotatedEvent.get("nodeId");
+          if (nodeId != null) {
+            handler.onLongPress(nodeId);
+          }
+          break;
+        }
+        case "tooltip": {
+          String tooltipMessage = (String) data.get("message");
+          handler.onTooltip(tooltipMessage == null ? "" : tooltipMessage);
```

Comment:
Also skip this message if it's missing the parameter

Reply:
Done